src/mozanalysis/sizing.py

```diff
@@ -162,6 +163,27 @@ def get_single_window_data(
                 "Either custom_target_query or target_list must be provided"
             )
 
+        # validate metric_list and target_list are from the same source
+        # filter out el.app_name is None, there are cases where metric-hub
+        # was not used and it's up to users to ensure the sources match
+        metric_list_sources = {el.app_name for el in metric_list if el.app_name}
+        target_list_sources = {el.app_name for el in target_list if el.app_name}
+
+        if len(metric_list_sources) > 1:
```

Comment:
We should probably raise warnings here, rather than errors, as I could see some cases where multiple app_names might be used -- for instance, joins to the metrics in multi-product

Reply:
That makes sense, fixed and updated the tests which had to modified to accommodate this